### ISSUE TITLE
Migration to Kotlin Multiplatform

### ISSUE THREAD

**Issue by kkostov** [pretix/pretixscan-desktop#72]
This issue tracks progress for the migration of the app to Kotlin Multiplatform.

## Preparations

- [X] Setup project and gradle structure
- [X] Migration plan for resources - images, translations
- [X] Add support for SQLite queries using existing storage and schema
- [X] Windows builds
- [] Pipeline setup for releases
- [] Document translation workflow using weblate

## Functional implementation
Each point can be tackled in two steps - implementing the feature functionally (bringing in all needed dependencies, writing out a compatible view model etc) and second step of refining the UI to meet our needs on desktop.

- [ ]  Disclaimer screen
- [ ]  Setup (connection and initialisation of the device)
- [ ] Welcome (input and submission of ticket codes)
    - [ ] Ticket validation feedback
    - [ ] Select event
    - [ ] Select check-in list
    - [ ] ...

**Comment by kkostov** [comment]
Testing feedback (Windows 11, 04/04/2025)

- [ ] On the main screen, it's unclear which buttons currently have keyboard focus (default accessibility is not sufficient -> focus colours should be more prominent)
  - [ ] On the main screen, after selecting a ticket search result with the keyboard, the result remains highlighted even when it looses focus (e.g. the next press of the Tab key focuses the Choose Event button, but a ticket result remains highlighted)
- [ ] Where are the application logs?
- [ ] Setup doesn't create a desktop shortcut by default. After installing the app, it's only available in C:\Program Files\pretixSCAN\...
- [ ] When pretixSCAN is launched, the NVIDIA Overlay popup shows up as if it's a game that's being started.

**Comment by rootjack404** [comment]
I know this branch is still WIP, and probably wont release in a short time, as I tried the built on macOS for my curiousity, I want to ask, does the auto select/scan with barcode scanner doesn't implemented yet, or the one extra click, after the QR code scanned is on purposes? Because I didn't see it in the Functional Implementation. Because in the stable version, we only need to scan, and the ticket automatically checked-in or checked-out without any extra click.

**Comment by kkostov** [comment]
Good catch @rootjack404, that's definitely a regression in recent commits. I will have a look!

- [x] Auto select/scan with barcode scanner -> we only need to scan, and the ticket automatically checked-in or checked-out without any extra click

**Comment by kkostov** [comment]
@rootjack404 the auto-scan issue should be addressed now.

**Comment by rootjack404** [comment]
@kkostov Great! I have tested it, and it work well, but seems like printing badge doesnt work yet on Mac, I haven't tried in Windows though since I don't have one in hand yet
